### PR TITLE
move Profiler API init

### PR DESCRIPTION
**Pull request by jpbempel**
Profiler API initialization before checking Profiler is enabled or
api key.

### CODE REVIEW

File: dd-java-agent/agent-profiling/src/main/java/com/datadog/profiling/agent/ProfilingAgent.java [hunk 1]
```diff
@@ -30,6 +30,8 @@
    */
   public static synchronized void run(final boolean isStartingFirst)
       throws IllegalArgumentException {
+    // init the Profiler API for the Tracer, no need to have the profiler enabled for that
+    Profiler.initialize(new JMXSessionFactory());
```
Comment (jbachorik) [inline dd-java-agent/agent-profiling/src/main/java/com/datadog/profiling/agent/ProfilingAgent.java:34]: Just a nit - we should perhaps rename `Profiler` to something else. It is rather confusing now - initializing Profiler even if profiler is not enabled :)
`MethodLeverTracer` for the lack of more witty names?

Comment (jpbempel) [reply dd-java-agent/agent-profiling/src/main/java/com/datadog/profiling/agent/ProfilingAgent.java:34]: I am on!

**Comment by mar-kolya** [comment]
Is there any reason to tie up MLT init with profiling at all?

**Comment by jbachorik** [comment]
AFAIK, just the convenience of being run at the correct time regarding the JMX hacks ... Otherwise we would need to copy/paste all the logic to MLT.

**Comment by mar-kolya** [comment]
I think we should not assume profiling will always use JMX - from that perspective we should split it... otherwise that is a hidden assumption that will bite us later.

**Comment by jbachorik** [comment]
Yes, ideally we should have a common scaffolding providing easy hooks to lifecycle. But we are not there yet and blocking MLT until we are there does not seem very feasible.

**Comment by mar-kolya** [comment]
FWIW I think 'hooks' necesary for this are already in place. They might not be ideal (a bit too much copy-paste) - but they are sufficient for the task...

**Comment by jbachorik** [comment]
https://datadoghq.atlassian.net/browse/PROF-1618

**Pull request merged**